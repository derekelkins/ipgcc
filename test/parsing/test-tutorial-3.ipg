Factor
  -> "(" Exp ")" { value = Exp.value }
   / Number { value = Number.value };

Number
  -> Digit { leadingDigit = Digit.value }
     ?[ leadingDigit != 0 ]
     repeat Digit.value
     { value = digitsToNum(leadingDigit, Digit.values) };

Digit
  -> "0" { value = 0 }
   / "1" { value = 1 }
   / "2" { value = 2 }
   / "3" { value = 3 }
   / "4" { value = 4 }
   / "5" { value = 5 }
   / "6" { value = 6 }
   / "7" { value = 7 }
   / "8" { value = 8 }
   / "9" { value = 9 };

