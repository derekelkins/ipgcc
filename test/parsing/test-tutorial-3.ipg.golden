Factor
  -> "("[0, 1]
     Exp@0[1, EOI]
     ")"[Exp@0.END, Exp@0.END + 1]
     { value = Exp@0.value }
   / Number@0[0, EOI]
     { value = Number@0.value };

Number
  -> Digit@0[0, EOI]
     { leadingDigit = Digit@0.value }
     ?[ leadingDigit != 0 ]
     repeat Digit@1[Digit@1.END, EOI].value starting on [Digit@0.END, EOI]
     { value = digitsToNum(leadingDigit, Digit@1.values) };

Digit
  -> "0"[0, 1]
     { value = 0 }
   / "1"[0, 1]
     { value = 1 }
   / "2"[0, 1]
     { value = 2 }
   / "3"[0, 1]
     { value = 3 }
   / "4"[0, 1]
     { value = 4 }
   / "5"[0, 1]
     { value = 5 }
   / "6"[0, 1]
     { value = 6 }
   / "7"[0, 1]
     { value = 7 }
   / "8"[0, 1]
     { value = 8 }
   / "9"[0, 1]
     { value = 9 };